sdk/mobile-center-crashes/src/test/java/com/microsoft/azure/mobile/crashes/CrashesTest.java

```diff
@@ -1170,4 +1179,41 @@ public void saveWrapperSdkErrorLogIOException() throws IOException, JSONExceptio
         verifyStatic();
         MobileCenterLog.error(anyString(), anyString(), any(IOException.class));
     }
+
+    @Test
+    public void sendMoreThan2ErrorAttachments() throws IOException, ClassNotFoundException, JSONException {
+        ArrayList<ErrorAttachmentLog> errorAttachmentLogs = new ArrayList<>(3);
+        errorAttachmentLogs.add(mock(ErrorAttachmentLog.class));
+        errorAttachmentLogs.add(mock(ErrorAttachmentLog.class));
+        errorAttachmentLogs.add(mock(ErrorAttachmentLog.class));
+
+        CrashesListener listener = mock(CrashesListener.class);
+        when(listener.shouldProcess(any(ErrorReport.class))).thenReturn(true);
+        when(listener.getErrorAttachments(any(ErrorReport.class))).thenReturn(errorAttachmentLogs);
+
+        LogSerializer logSerializer = mock(LogSerializer.class);
+        when(logSerializer.deserializeLog(anyString())).thenReturn(mock(ManagedErrorLog.class));
+
+        mockStatic(ErrorLogHelper.class);
+        when(ErrorLogHelper.getLastErrorLogFile()).thenReturn(mock(File.class));
+        when(ErrorLogHelper.getStoredErrorLogFiles()).thenReturn(new File[]{mock(File.class)});
+
+        when(StorageHelper.InternalStorage.read(any(File.class))).thenReturn("");
+        when(StorageHelper.InternalStorage.readObject(any(File.class))).thenReturn(mock(Throwable.class));
+
+        Crashes crashes = Crashes.getInstance();
+        crashes.setInstanceListener(listener);
+        crashes.setLogSerializer(logSerializer);
+
+        Crashes spy = spy(crashes);
```

Comment:
You can't spy on this as getInstance will return the original object and not the spy.
You can instead use `whenNew` to mock the new ErrorReport that gets created in Crashes.
There is plenty of `whenNew` examples in our project for reference.